dotnet/src/PF-Extensions/Planning.ChatStepwisePlanner/ChatStepwisePlanner.cs

```diff
@@ -0,0 +1,539 @@
+
+        this._kernel = kernel ?? throw new ArgumentNullException(nameof(kernel));
+
+        this.Config = config ?? new();
+        this.Config.ExcludedSkills.Add(RestrictedSkillName);
+
+        var promptConfig = new PromptTemplateConfig();
+        var systemPromptTemplate = systemPrompt ?? EmbeddedResource.Read("Microsoft.SemanticKernel.Planning.ChatStepwise.Prompts.systemprompt.txt");
+        var fewShotExamplesTemplate = fewShotExamples ?? JsonSerializer.Deserialize<FewShotExample[]>(EmbeddedResource.Read("Microsoft.SemanticKernel.Planning.ChatStepwise.Prompts.fewshotexamples.json"));
+        string promptConfigString = EmbeddedResource.Read("Microsoft.SemanticKernel.Planning.ChatStepwise.Prompts.config.json");
+        if (!string.IsNullOrEmpty(promptConfigString))
+        {
+            promptConfig = PromptTemplateConfig.FromJson(promptConfigString);
+        }
+
+        promptConfig.Completion.MaxTokens = this.Config.MaxTokens;
+
+        //this._systemStepFunction = this.ImportSemanticFunction(this._kernel, "StepwiseStep", promptTemplate, promptConfig);
+        this._nativeFunctions = this._kernel.ImportSkill(this, RestrictedSkillName);
+
+        this._context = this._kernel.CreateNewContext();
+        this._logger = this._kernel.Log;
+    }
+
+    public Plan CreatePlan(string goal)
+    {
+        if (string.IsNullOrEmpty(goal))
+        {
+            throw new PlanningException(PlanningException.ErrorCodes.InvalidGoal, "The goal specified is empty");
+        }
+
+        string functionDescriptions = this.GetFunctionDescriptions();
+
+        Plan planStep = new(this._nativeFunctions["ExecutePlan"]);
+        planStep.Parameters.Set("functionDescriptions", functionDescriptions);
+        planStep.Parameters.Set("question", goal);
+
+        //planStep.Outputs.Add("agentScratchPad");
+        planStep.Outputs.Add("stepCount");
+        planStep.Outputs.Add("skillCount");
+        planStep.Outputs.Add("stepsTaken");
+
+        Plan plan = new(goal);
+
+        plan.AddSteps(planStep);
+
+        return plan;
+    }
+
+    [SKFunction, SKName("ExecutePlan"), Description("Execute a plan")]
+    public async Task<SKContext> ExecutePlanAsync(
+        [Description("The question to answer")]
+        string question,
+        [Description("List of tool descriptions")]
+        string functionDescriptions,
+        SKContext context)
+    {
+        var stepsTaken = new List<SystemStep>();
+        this._logger?.BeginScope("ChatStepwisePlanner");
+        if (!string.IsNullOrEmpty(question))
+        {
+            this._logger?.LogInformation("Ask: {Question}", question);
+
+            // TODO MOVE
+
+            var systemPromptTemplate = EmbeddedResource.Read("Microsoft.SemanticKernel.Planning.ChatStepwise.Prompts.systemprompt.txt");
+            var fewShotExamplesTemplate = JsonSerializer.Deserialize<FewShotExample[]>(EmbeddedResource.Read("Microsoft.SemanticKernel.Planning.ChatStepwise.Prompts.fewshotexamples.json"));
```

Comment:
What's the reason for having the few shot example as a separate file that's not included in the prompt template? Would we add multiple specific examples in the future?

Reply:
yes, I can imagine adding question specific examples while keeping the same system prompt.